test/xskbench/xskbench.c

```diff
@@ -1834,13 +1870,72 @@ main(
 
     for (UINT32 tIndex = 0; tIndex < threadCount; tIndex++) {
         MY_THREAD *Thread = &threads[tIndex];
-        WaitForSingleObject(Thread->threadHandle, INFINITE);
+        PlatWaitThread(&Thread->threadHandle, INFINITE);
         for (UINT32 qIndex = 0; qIndex < Thread->queueCount; qIndex++) {
-            PrintFinalStats(&Thread->queues[qIndex]);
+            MY_QUEUE *Queue = &Thread->queues[qIndex];
+#if defined(_KERNEL_MODE)
+            KFLOATING_SAVE FloatingSave;
+            if (KeSaveFloatingPointState(&FloatingSave) == STATUS_SUCCESS) {
+#endif
+                PrintFinalStats(Queue);
+#if defined(_KERNEL_MODE)
+                KeRestoreFloatingPointState(&FloatingSave);
+            }
+#endif
```

Comment:
Kernel has to specifically configure to use floating point operations?!

Reply:
Yeah, I thought this was well known :)

You have to save the floating point registers if you're going to do floating point, because they aren't automatically saved/restored upon context switch. You can't trash (and leak) the registers back to user mode.

Reply:
TIL. Thanks.